lib/web_ui/lib/src/engine/html/shader_mask.dart

```diff
@@ -0,0 +1,304 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// @dart = 2.12
+part of engine;
+
+/// A surface that applies a shader to its children.
+///
+/// Currently there are 2 types of shaders:
+///   - Gradients
+///   - ImageShader
+///
+/// Gradients
+///   The gradients can be applied to the child tree by rendering the gradient
+///   into an image and referencing the image in an svg filter to apply
+///   to DOM tree.
+///
+class PersistedShaderMask extends PersistedContainerSurface
+    implements ui.ShaderMaskEngineLayer {
+  PersistedShaderMask(
+      PersistedShaderMask? oldLayer,
+      this.shader,
+      this.maskRect,
+      this.blendMode,
+      ) : super(oldLayer);
+
+  html.Element? _childContainer;
+  final ui.Shader shader;
+  final ui.Rect maskRect;
+  final ui.BlendMode blendMode;
+  html.Element? _shaderElement;
+  html.Element? _imageElement;
```

Comment:
I only see this element being removed and set to null? Is it used for anything?

Reply:
Was for future ImageShader. removed for now.